### PR TITLE
DOCSP-33481: Update BSONOptions Example

### PR DESCRIPTION
**Pull request by norareidy**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-33481>
Staging - <https://preview-mongodbnorareidy.gatsbyjs.io/golang/DOCSP-33481-bson-options-example/fundamentals/bson/>

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:62]: S: use source constant where possible

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:236]: S: follow convention 
```suggestion
       ApplyURI("<connection string>").
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:218]:
```suggestion
This example specifies the following ``BSONOptions`` configuration:
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:223]:
```suggestion
- Sets ``UseJSONStructTags`` to ``true``, which instructs the driver to use the ``"json"``
  struct tag if a ``"bson"`` struct tag is not specified
- Sets ``DefaultDocumentD`` to ``true``, which instructs the driver to always unmarshal documents
  into the ``primitive.D`` type
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:241]: I: this tip does not actually link to the BSONOptions API documentation. Can you rewrite this tip so that it links to the API documentation and the BSONOptions example (making link text clear though eg To learn more about the type, see api documentation. To see an example, <link to example>)

**Review by rustagir — CHANGES_REQUESTED** [review]
just a few things but great job cleaning up this section!

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:57]:
```suggestion
types. The {+driver-short+} can marshal and unmarshal structs and other native Go
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:58]: I: not sure its convention to use slashes
```suggestion
types to and from BSON using a `configurable codec system <{+api+}/bson/bsoncodec>`_.
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:61]:
```suggestion
{+driver-short+} by using **struct tags**, which are optional pieces of metadata
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:65]:
```suggestion
use in the {+driver-short+}:
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:93]:
```suggestion
If you don't specify struct tags, the {+driver-short+} marshals
structs by using the following rules:
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:246]:
```suggestion
   For an example that specifies a ``BSONOptions`` instance and creates a client with these options, see the
   `Connect() BSONOptions example <{+api+}/mongo#example-Connect-BSONOptions>`__.
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:215]:
```suggestion
You can create a ``BSONOptions`` instance to set BSON options on your ``Client``.
```

File: source/fundamentals/bson.txt
Comment (rustagir) [inline source/fundamentals/bson.txt:223]:
```suggestion
This example performs the follow actions when configuring a ``BSONOptions`` instance:

- Sets the ``UseJSONStructTags`` field to ``true``, which instructs the driver to
  use the ``json`` struct tag if a ``bson`` struct tag is not specified.
- Sets the ``DefaultDocumentD`` field to ``true``, which instructs the driver to
  always unmarshal documents into the ``primitive.D`` type.
```

**Review by rustagir — CHANGES_REQUESTED** [review]
a few more things - let me know if you have any questions

**Review by rustagir — APPROVED** [review]
approving with some final comments, let me know if you want another review anyways though!

File: source/fundamentals/bson.txt
Comment (matthewdale) [inline source/fundamentals/bson.txt:61]: This kinda makes it sound like the only way to modify BSON marshaling and unmarshaling behavior is using struct tags. However, users can also use `BSONOptions` to modify default marshaling/unmarshaling behavior. Generally struct tags are for modifying the marshaling/unmarshaling behavior of a single struct field and `BSONOptions` are for modifying the marshaling/unmarshaling behavior of an entire `Client`. Is there a way to clarify that?

File: source/fundamentals/bson.txt
Comment (matthewdale) [inline source/fundamentals/bson.txt:225]: This option will become the default behavior in Go Driver 2.0, so it may not be the best example to use if forward-compatibility is a concern. Is there a specific motivation to demonstrate setting `DefaultDocumentD`?

File: source/fundamentals/bson.txt
Comment (norareidy) [reply source/fundamentals/bson.txt:58]: Good to know, I've taken that sentence out

File: source/fundamentals/bson.txt
Comment (norareidy) [reply source/fundamentals/bson.txt:61]: Edited to clarify this (both here and in the BSON options section)

File: source/fundamentals/bson.txt
Comment (norareidy) [reply source/fundamentals/bson.txt:225]: Thanks for this information! There was no specific reason for choosing DefaultDocumentD, I just wanted to show a couple different BSONOptions fields. I've changed this field to the OmitZeroStruct field if that works better.

**Review by matthewdale — APPROVED** [review]
Looks good 👍

**Pull request merged**
